### PR TITLE
YLW 3252 Create local web API server

### PR DESCRIPTION
**Pull request by davidgreenland**
#### What 
Created a local API that serves up data from the agify.io and api.wherestheiss.at

#### Why
To make available several endpoints that connect to the external apis and then return the data.

#### How
Used ASP.NET web api to create the API.
Made endpoints at available at 
http://localhost:5005/api/satellite/iss 
http://localhost:5005/api/agify

### CODE REVIEW

File: DGWebApi/Controllers/AgifyController.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public class AgifyController : ControllerBase
+    {
+        [HttpGet("{name}")]
+        public async Task<IActionResult> GetByName(string name, IAgifyService agifyService)
+        {
+            var nameCount = await agifyService.GetName(name);
+
+            if (nameCount == null)
+            {
+                return StatusCode(500);
```
Comment (patrickfallon) [inline DGWebApi/Controllers/AgifyController.cs]: You don't want to intentionally return 500 errors. 500 means an unexpected error so it shouldn't be returned as normal flow

File: DGWebApi/Services/AgifyService.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+        return JsonSerializer.Deserialize<NameCount>(json);
+    }
+
+    public async Task<NameCount?> GetName(string name)
+    {
+        var response = await _httpClient.GetAsync($"?name={name}");
+        var json = await response.Content.ReadAsStringAsync();
+
+        return JsonSerializer.Deserialize<NameCount>(json);
```
Comment (patrickfallon) [inline DGWebApi/Services/AgifyService.cs:30]: Is it not possible to see a response where count is 0?

File: DGWebApi/Utils/LocationUtils.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Globalization;
+
+namespace DGWebApi.Utils
+{
+    public class LocationUtils
+    {
+        public static string GetNameFromCountryCode(string code)
+        {
+            var cultureInfo = new CultureInfo(code);
+            var ri = new RegionInfo(cultureInfo.Name);
```
Comment (patrickfallon) [inline DGWebApi/Utils/LocationUtils.cs]: As discussed, this throws an error when incorrect codes are used

File: DGWebApi/Services/AgifyService.cs (hunk 2 above)
Comment (davidgreenland) [reply DGWebApi/Services/AgifyService.cs:30]: sorted with nullable in the model

File: DGWebApi/Utils/LocationUtils.cs (hunk 3 above)
Comment (davidgreenland) [reply DGWebApi/Utils/LocationUtils.cs]: I've added a try catch block. Let me know if there is a better way to handle this please.

File: DGWebApi/Controllers/AgifyController.cs (hunk 1 above)
Comment (davidgreenland) [reply DGWebApi/Controllers/AgifyController.cs]: I've made it 404.

File: DGWebApi/Controllers/SatelliteController.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
+﻿using DGWebApi.Models;
+using Microsoft.AspNetCore.Mvc;
+using DGWebApi.Services.Interfaces;
+
+namespace DGWebApi.Controllers;
```
Comment (patrickfallon) [inline DGWebApi/Controllers/SatelliteController.cs:5]: Choose either block-scoped or file-scoped namespace definitions - each controller is using a different style

**Review by patrickfallon — APPROVED** [review]

File: DGWebApi/Controllers/SatelliteController.cs (hunk 4 above)
Comment (davidgreenland) [reply DGWebApi/Controllers/SatelliteController.cs:5]: done

**Pull request merged**